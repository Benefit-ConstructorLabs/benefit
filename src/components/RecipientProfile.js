import React from 'react';
import PropTypes from 'prop-types';
import AnimationWrapper from './AnimationWrapper';

const RecipientProfile = ({ profile }) => {
  const { firstName, lastName, photo, username, tel, bio, reason } = profile;
  return (
    <section className="recipient__profile">
      <img className="recipient__profile__photo" alt="recipient" src={photo} />
      <h3 className="recipient__profile__title">Your details</h3>
      <dl>
        <dt>Name</dt>
        <dd>{`${firstName} ${lastName}`}</dd>
        <dt>Username</dt>
        <dd>{username}</dd>
        <dt>Telephone</dt>
        <dd>{tel}</dd>
        <dt>Reason</dt>
        <dd>{reason}</dd>
        <dt>Your bio</dt>
        <dd>
          {bio.length > 0
            && (
              <ul>
                {bio.map(item => (<li key={`bio_${Math.random() * item.length}`}>{`${item}`}</li>))}
              </ul>
            )
          }
        </dd>
      </dl>

    </section>
  );
};

RecipientProfile.propTypes = {
  profile: PropTypes.instanceOf(Object).isRequired,
};

export default AnimationWrapper(RecipientProfile);
