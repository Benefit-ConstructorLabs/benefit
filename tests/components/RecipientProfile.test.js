import React from 'react'
import RecipientProfile from '../../src/components/RecipientProfile';
import renderer from 'react-test-renderer';

describe('App', () => {
  test('should render correctly', () => {
    const tree = renderer
      .create(<RecipientProfile />)
      .toJSON()
    expect(tree).toMatchSnapshot();
  });
})