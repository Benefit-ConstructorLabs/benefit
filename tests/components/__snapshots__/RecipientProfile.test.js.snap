// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipientProfile should render correctly 1`] = `
<div
  className="recipient-profile"
>
  <h3
    className="recipient-profile__headline"
  >
    Fancy a chat?
  </h3>
  <p
    className="recipient-profile__intro"
  >
     
    is interested in:
  </p>
  <ol
    className="recipient-profile__interests"
  >
    <li
      className="recipient-profile__interest"
    />
  </ol>
</div>
`;
